packages/core/src/bodl/bodl-emitter-service.ts

```diff
@@ -241,6 +241,18 @@ export default class BodlEmitterService implements BodlService {
             ...lineItems.physicalItems,
             ...lineItems.digitalItems,
         ].map((item) => {
+            const getCategoryNames = (lineItem: LineItem): string[] => {
+                if (Array.isArray(lineItem.categoryNames)) {
+                    return lineItem.categoryNames;
+                } else if (Array.isArray(lineItem.categories)) {
+                    return Array.prototype.concat
+                        .apply([], lineItem.categories)
+                        .map(({ name }) => name);
+                }
```

Comment:
One question why do we need `apply` here? Can we not map over `lineItem.categories` directly?

Reply:
`lineItems.categories` is an array of with nested arrays of objects. It looks something like `[[{ name: "Shop All" }], [{ name: "Kitchen" }]]` when we get it from the API. So with this I first concat them into one flat array of category objects and then map the result array to have only names

Reply:
Interesting since this https://github.com/bigcommerce/checkout-sdk-js/blob/44fa6198236c5492df3c9bffabbc769a474c0fb5/packages/core/src/payment/strategies/affirm/affirm-payment-strategy.ts#L314 just maps over a list. 

So for above case can we use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat or a lodash method.

Reply:
@animesh1987,
Updated. I previously intended to use the `flat` method however, I would need to change `compilerOptions` and replace `es6` with `es2019`. I didn't feel confident doing it, so used that old school approach. I didn't think about lodash though. Used it now